core/miner_operation.go

```diff
@@ -81,6 +81,9 @@ func (op *stakeAddOp) Validate() error {
 	if !op.msg.Amount().IsUint64() {
 		return fmt.Errorf("amount type not uint64")
 	}
+	if *op.msg.Operator() != *op.msg.OpTarget() && op.minerType == types.MinerTypeVerify{
```

Comment:
Should check at Operation() stage so that the gas was consumed in case of ddos attack